src/linker/pipeline.py

```diff
@@ -78,6 +78,9 @@ def _get_implementations(self) -> Tuple[Implementation, ...]:
             Implementation(
                 step.name,
                 self.config.pipeline["steps"][step.name]["implementation"],
+                self.config.pipeline["steps"][step.name].get(
+                    "implementation_configuration", None
```

Comment:
`implementation_configuration` seems redundant, since it's within the specifications of an implementation. Can we not leave the key as `configuration` or something similar?

Reply:
It's not, it's within the specification of the pipeline steps. See `pipeline_2_0_0.yaml`

Reply:
I did look at that and I still think including "implementation" in the key is redundant.

Reply:
Hmm maybe it would be better to nest it under "implementation" like so?

```yaml
step_1:
  implementation:
    name: step_1_python_pandas
    configuration:
      foo: bar
```

instead of

```yaml
step_1:
  implementation: step_1_python_pandas
  implementation_configuration: 
    foo: bar
```

Or, I think Rajan's idea of just calling it "configuration" could work too. I don't think it's fully _redundant_ but it might be unambiguous even without "implementation."

Reply:
I think nesting makes the most sense

Reply:
Will implement and push to this PR